### ISSUE TITLE
[enhancement request] Please add checkbox to enable/disable

### ISSUE THREAD

**Issue by lnlp** [jneilliii/OctoPrint-MultiLineTerminal#4]
This is a useful plugin but only for the moments that you want to send multiple commands/lines at once. While for most situations this is not the case.

A disadvantage of this plugin is that it disables up/down arrow key functionality to restore a previous commandline so that you don't have to type it again.

It would be nice if we could have a small checkbox next to the input field to enable/disable multi-line functionality so that for most cases when only a single line is needed the up/down arrows can still be used to restore previously entered commands and that multi-line can be quickly enabled (and disabled) only when needed.

For now I have uninstalled the plugin again because it disables up/down key restore functionality.

**Comment by jneilliii** [comment]
It actually does support command history by using shift+up and shift+down.

**Comment by lnlp** [comment]
Thanks. I wasn't aware of that.
I now see that it is documented in the README.md on GitHub:

> This plugin simply replaces the single line gcode input into a multi-line text area. Once installed you can use the Send button or shift+enter to send the gcode commands to your printer. Use shift+up and shift+down to navigate through command history.

But on https://plugins.octoprint.org/plugins/multilineterminal that information is missing:
_(This is what is shown when clicking on 'Details' in the 'Add new plugins' dialog.)_

> This plugin simply replaces the single line gcode input into a multi-line text area. Once installed you will have to use the Send button instead of the Enter key to send the commands to your printer.

It will be helpful to update the latter.

---

From UX perspective it may still be useful to add the suggested checkbox because for quick single line actions (which are probably most commonly used) having to press only Enter is quicker and easier than having to press shift-Enter.
With shift-up and shift-down functionality already in place there is less necessity for the checkbox, but nice to have.

**Comment by jneilliii** [comment]
Yeah, I need to update the plugin repo listing as that feature was just recently added. I will consider the toggle option for a future update.

**Comment by jneilliii** [comment]
I just pushed an update to the plugin to an 0.1.5 branch that will load the interface in single line mode by default and work like the standard octoprint interface and then you can switch to multi-line mode using shift enter from single line mode or double clicking the input field. Input should grow and shrink based on the number of lines in the input. If you want to test it out for me and verify it's working as intended, you can install by copying/pasting the URL below in Plugin Manager > Get More > ...fromURL and clicking install.

```
https://github.com/jneilliii/OctoPrint-MultiLineTerminal/archive/0.1.5.zip
```

**Comment by jneilliii** [comment]
I just released this as version 0.1.5 that allows for toggling between single line mode (default) and multiline mode with either shift+enter or double-click of input field.
